Bottom: c05cbbc35f8090fd5b40006bedcfbdda85cf7e06
Top:    cf5d3a146245482c3c0d60d661861483422ac9c6
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 00:52:28 +0100

Refresh of draw-c-split-gathering-routine-2

---

diff --git a/src/draw.c b/src/draw.c
index 09c2715..41561f8 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -194,17 +194,6 @@ Redraw (void)
   gui->invalidate_all ();
 }
 
-static int
-element_callback (const BoxType * b, void *cl)
-{
-  ElementTypePtr element = (ElementTypePtr) b;
-  int *side = cl;
-
-  if (ON_SIDE (element, *side))
-    DrawElementPackage (element);
-  return 1;
-}
-
 static void
 draw_element_name (ElementType *element)
 {
@@ -339,6 +328,17 @@ draw_element_package (ElementType *element)
   END_LOOP;
 }
 
+static int
+element_callback (const BoxType * b, void *cl)
+{
+  ElementTypePtr element = (ElementTypePtr) b;
+  int *side = cl;
+
+  if (ON_SIDE (element, *side))
+    DrawElementPackage (element);
+  return 1;
+}
+
 /* ---------------------------------------------------------------------------
  * prints assembly drawing.
  */
@@ -1754,7 +1754,7 @@ EraseElementName (ElementTypePtr Element)
 
   if (TEST_FLAG (HIDENAMEFLAG, Element))
     return;
-  AddPart (&ELEMENT_TEXT (PCB, Element));
+  DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), PCB->minSlk);
 }
